### PR TITLE
Prepare 0.100.7 release

### PR DESCRIPTION
**Pull request by alejoe91**
Propagated the following bug-fix PRs:

- #2649 
- #2810
- #2828
- #2889
- #2891

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Guys, can we setup the correct version on the pyproject? See:
https://github.com/SpikeInterface/spikeinterface/pull/2834

File: pyproject.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ dependencies = [
     "numpy",
     "threadpoolctl>=3.0.0",
     "tqdm",
-    "zarr>=0.2.16",
+    "zarr>=0.2.16, <2.18",
```
Comment (alejoe91) [inline pyproject.toml:26]:
```suggestion
    "zarr>=2.16, <2.18",
```

**Comment by h-mayorquin** [comment]
Would there be a release on probeinterface? If not, could we also add this so we can pin neuroconv to the incoming release:

https://github.com/SpikeInterface/spikeinterface/pull/2833

Also, if you feel like indulging me, having this release will make my life in the following month easier:


https://github.com/SpikeInterface/spikeinterface/pull/2969

**Comment by alejoe91** [comment]
> Would there be a release on probeinterface? If not, could we also add this so we can pin neuroconv to the incoming release:
> 
> #2833
> 
> Also, if you feel like indulging me, having this release will make my life in the following month easier:
> 
> #2969

Both are new features that were not there in 0.100.* ;)

**Pull request merged**